### PR TITLE
Mapping; fixes #6

### LINKED ISSUE #6: DST-ECO-SR-0010: Mapping Rules
**DST-ECO-SR-0010**

As a minimum the ECOSIM PRO adapter shall support exchange of the following [AD3] Annex A + B information between Hub and DST: DomainOfExpertise, Option (for one selected Option), ElementDefinition, ElementUsage, NestedElement, all Parameters with values, associated ActualFiniteStateList.

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/pulls) open
- [x] I have verified that I am following the DEHP-EcosimPro [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-ECOSIMPRO/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
fixes #6 

- [x] Added the ability to map data from EcosimPro to 10-25
    - [x] Added the mapping rule that allows it
- [x] Context menu added
- [x] Switch panel / mapping direction feature

### CODE REVIEW

File: DEHPEcosimPro/DstController/DstController.cs [hunk 1]
```diff
@@ -83,6 +96,20 @@ public bool IsSessionOpen
         /// </summary>
         public int RefreshInterval => this.opcClientService.RefreshInterval;
 
+        /// <summary>
+        /// Backing field for the <see cref="MappingDirection"/>
+        /// </summary>
+        private MappingDirection mappingDirection;
```
Comment (ahmedatrhea) [inline DEHPEcosimPro/DstController/DstController.cs:102]: private fields are grouped up in the file

File: DEHPEcosimPro.Tests/ViewModel/Rows/VariableRowViewModelTestFixture.cs [hunk 2]
```diff
@@ -95,21 +94,21 @@ public void VerifyAverageCalculation()
...
+            var newValues = new List<TimeTaggedValueRowViewModel>()
             {
-                (131234, DateTime.MinValue.AddDays(1)), 
-                (-143298.5224323, DateTime.MinValue.AddDays(1)),
-                (2u, DateTime.MinValue.AddDays(1)),
-                (44.87613, DateTime.MinValue.AddDays(1)),
-                (0.42e2, DateTime.MinValue.AddDays(1)),
-                (.12387, DateTime.MinValue.AddDays(1)),
-                (2ul, DateTime.MinValue.AddDays(1))
+                new TimeTaggedValueRowViewModel(131234, DateTime.MinValue.AddDays(1)),
```
Comment (alexatrhea) [inline DEHPEcosimPro.Tests/ViewModel/Rows/VariableRowViewModelTestFixture.cs:99]: This is I guess placeholder for when SampledFunctionParameterType makes its appearance.

File: DEHPEcosimPro/Views/MainWindow.xaml [hunk 3]
```diff
@@ -118,8 +123,8 @@
                     <ProgressBar Background="AliceBlue" Foreground="DarkGray" Maximum="100" Minimum="0" Value="37" />
                     <TextBlock HorizontalAlignment="Center" VerticalAlignment="Center" FontSize="18" Text="37%" />
                 </Grid>
-                <Button Grid.Row="1" Grid.Column="0" Height="40" Margin="10,5,5,10" Content="Cancel"/>
-                <Button Grid.Row="1" Grid.Column="1" Height="40" Margin="5,5,10,10" Content="Map"/>
+                <Button Grid.Row="1" Grid.Column="0" Height="40" Margin="10,5,5,10" Content="CANCEL"/>
+                <Button Grid.Row="1" Grid.Column="1" Height="40" Margin="5,5,10,10" Content="TRANSFERT"/>
```
Comment (alexatrhea) [inline DEHPEcosimPro/Views/MainWindow.xaml:127]: There is a type in "Transfer"

Comment (alexatrhea) [reply DEHPEcosimPro/Views/MainWindow.xaml:127]: Also I think you can keep it "Cancel" and "Transfer"

**Review by alexatrhea — CHANGES_REQUESTED** [review]
small changes requested

File: DEHPEcosimPro/DstController/DstController.cs [hunk 4]
```diff
@@ -230,5 +269,30 @@ public void CloseSession()
...
+            this.ElementDefinitionParametersDstVariablesMaps = elements;
+            this.ExternalIdentifierMaps = maps;
+            return true;
+        }
+
+        /// <summary>
+        /// Transfers the mapped variables to the Hub data source
+        /// </summary>
+        /// <returns>A <see cref="Task"/></returns>
+        public async Task Transfert()
```
Comment (ahmedatrhea) [inline DEHPEcosimPro/DstController/DstController.cs:292]: typo in "Transfer"

File: DEHPEcosimPro/Views/MainWindow.xaml [hunk 5]
```diff
@@ -98,7 +103,7 @@
 
                                 <ListView x:Name="Mapping" Grid.Column="0" HorizontalContentAlignment="Stretch" ItemTemplate="{StaticResource MappedThingRowDataTemplate}" />
                                 <Button Grid.Column="1" Margin="10" Background="Transparent" BorderBrush="Transparent">
-                                    <TextBlock FontSize="18" Text="🕑 Mapping History" TextWrapping="Wrap" TextAlignment="Center"/>
+                                    <TextBlock FontSize="18" Text="🕑 Transfert History" TextWrapping="Wrap" TextAlignment="Center"/>
```
Comment (ahmedatrhea) [inline DEHPEcosimPro/Views/MainWindow.xaml:106]: typo in "Transfer"

**Review by ahmedatrhea — APPROVED** [review]
approved with minor changes

File: DEHPEcosimPro.Tests/ViewModel/Rows/VariableRowViewModelTestFixture.cs (hunk 2 above)
Comment (nathanatrhea) [reply DEHPEcosimPro.Tests/ViewModel/Rows/VariableRowViewModelTestFixture.cs:99]: Yes indeed

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**